libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRouteLine.kt

```diff
@@ -114,12 +121,25 @@ class CarRouteLine internal constructor(
             locationPlugin.addOnIndicatorPositionChangedListener(onPositionChangedListener)
             mainCarContext.mapboxNavigation.registerRouteProgressObserver(routeProgressObserver)
             routesProvider.registerRoutesListener(routesListener)
+
+            styleScope.coroutineContext.cancelChildren()
+            styleScope.launch {
+                mainCarContext.routeAlternativesEnabled.collect { enabled ->
+                    if (enabled) {
+                        routeLineView.showAlternativeRoutes(style)
+                    } else {
+                        routeLineView.hideAlternativeRoutes(style)
+                    }
+                }
+            }
         }
     }
 
     override fun onDetached(mapboxCarMapSurface: MapboxCarMapSurface) {
         logAndroidAuto("CarRouteLine carMapSurface detached $mapboxCarMapSurface")
         val mapSurface = mapboxCarMapSurface.mapSurface
+        mapboxCarMapSurface.handleStyleOnDetached(styleLoadedListener)
+        styleScope.coroutineContext.cancelChildren()
```

Comment:
If you do it this way `styleScope.coroutineContext.cancelChildren()`. I would make it the very last statement in `onDetached`. one of the registered observers could launch a coroutine, and then it would be leaked.

the safer way is just, construct a new one on attach and cancel the scope so no more subscriptions are handled with `styleScope.cancel()`